src/main/java/entities/CommonUser.java

```diff
@@ -14,7 +14,7 @@ public class CommonUser implements User
     public CommonUser(int uid, String name, int status)
     {
         this.uid = uid;
-        userProfile = new UserProfile();
+        userProfile = new UserProfile(uid,name);
```

Comment:
if every user is storing a profile, it is redundant to store the UID in the profile